lib/test.py

```diff
@@ -341,8 +341,15 @@ def tap_parse_print_yaml(self, yml):
 
     def check_tap_output(self):
         """ Returns is_tap, is_ok """
-        with open(self.tmp_result, 'r') as f:
-            content = f.read()
+        try:
+            with open(self.tmp_result, 'r') as f:
+                content = f.read()
+        except UnicodeDecodeError as e:
+            color_stdout('\nCannot read result file (%s).\n' % str(e),
```

Comment:
It might be nice to write something about “maybe you are using binary” or “try to use base64” ... Up to you.

Reply:
The actual message looks so:

> Cannot read result file ('utf-8' codec can't decode byte 0xc2 in position 27: invalid continuation byte).

However I would highlight that we only unable to use it as TAP13: file-to-file comparison still work. The 'TAP13 parse failed' message below looks good here (and also suggest `--update-result`). So, maybe, just move open+read under the try-except block and catch UnicodeDecodeError in the except clause?

Reply:
BTW, it would be good to highlight the fact that the change is only about parsing output as TAP13 somewhere in the commit message. Now it is misleading a bit.

Reply:
@Totktonada fixed

Reply:
Thanks!